include/superior_mysqlpp/prepared_statements/prepared_statement_base.hpp

```diff
@@ -68,23 +94,39 @@ namespace SuperiorMySqlpp
             }
 
         protected:
+            /**
+             * This method implements the store() specialized behavior.
+             */
             void storeOrUse()
             {
                 this->statement.storeResult();
             }
         };
 
+        /**
+         * Specialization of StatementBase where statement results are "used" - not
+         * fetched all at one.
+         * Contains methods that are useful for progressive fetching of results.
+         */
         template<>
         class StoreOrUseResultBase<false> : public StatementBase
         {
         public:
             using StatementBase::StatementBase;
 
-
+            /**
+             * This method implements the use() specialized behavior.
+             * As this is the default mode for mysql client, nothing needs to be done.
+             */
             void storeOrUse() const
             {
             }
 
+            /**
+             * Returns the amount of fetched rows.
+             * @return Amount of already fetched rows.
+             * !!!!! Underlying mysql_stmt_num_rows does NOT gurantee such behavior!
```

Comment:
This returns already fetched rows, not total result rows, or even this lighter guarantee isn't right?

Reply:
It is not guaranteed explicitly in documentation. Further research needs to be made into source code if such behaviour is explicitly mentioned at least there or if it is purely implemntation detail.

Reply:
I did some testing, as basic tests did not cover this.
After calling this method in use() mode, it always returns 0 for me (with debian-jessie default mysql client). Was that ever tested at all?

I'll add suggestion for removal.

Reply:
Yes, this should probably be then counted manually in use spec.